packages/daemon/src/rpc/methods/comms.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+
+import {
+  type CommsGetSubscribableParams,
+  type CommsGetSubscribableResult,
+  type CommsPublishParams,
+  type CommsSubscribeParams,
+  COMMS_GET_SUBSCRIBABLE_SCHEMA,
+  COMMS_PUBLISH_SCHEMA,
+  COMMS_SUBSCRIBE_SCHEMA,
+} from '@mainframe/client'
+import { idType as fromClientID } from '@mainframe/utils-id'
+import type { Subscription as RxSubscription, Observable } from 'rxjs'
+
+import { pollFeedJSON } from '../../swarm/feed'
+import { ContextSubscription } from '../../context/ContextSubscriptions'
+import type ClientContext from '../../context/ClientContext'
+import type { NotifyFunc } from '../../rpc/handleClient'
+
+const MINUTE = 60 * 1000
+
+class CommsSubscription extends ContextSubscription<RxSubscription> {
+  constructor() {
+    super('comms_subscription')
+  }
+
+  startNotify(notify: NotifyFunc, observable: Observable<Object>) {
+    this.data = observable.subscribe(result =>
+      notify(this.method, { subscription: this.id, result }),
+    )
+  }
+
+  async dispose() {
+    if (this.data != null) {
+      this.data.unsubscribe()
+    }
+  }
+}
+
+export const publish = {
+  params: COMMS_PUBLISH_SCHEMA,
+  handler: async (
+    ctx: ClientContext,
+    params: CommsPublishParams,
+  ): Promise<void> => {
+    await ctx.mutations.publishAppData(
+      fromClientID(params.appID),
+      fromClientID(params.contactID),
+      params.key,
+      params.value,
+    )
+  },
+}
+
+export const subscribe = {
+  params: COMMS_SUBSCRIBE_SCHEMA,
+  handler: async (
+    ctx: ClientContext,
+    params: CommsSubscribeParams,
+  ): Promise<string> => {
+    const appFeeds = await ctx.queries.getContactAppFeeds(
+      fromClientID(params.appID),
+      fromClientID(params.contactID),
+    )
+    if (!appFeeds[params.key]) {
+      throw new Error('Feed not found')
+    }
+
+    const sub = new CommsSubscription()
+    ctx.subscriptions.set(sub)
+
+    const poll = pollFeedJSON(ctx.io.bzz, appFeeds[params.key].feedHash, {
```

Comment:
In cases the app calls `subscribe` multiple times with the same key, there is no de-duplication or multiplexing, is there?

Reply:
Nope but it should. I'll add a TODO since it's non-essential but desirable